File: Tutorial5-2Dagger-Android-SubComponents3-ContributesAndroidInjector/src/main/java/com/example/tutorial5_2dagger_android_subcomponents3/ToastMaker.java
package com.example.tutorial5_2dagger_android_subcomponents3;

import android.content.Context;
import android.widget.Toast;

public class ToastMaker {

    private Context context;

    public ToastMaker(Context context) {
        this.context = context;
    }

    public void showToast(String message) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
    }


}

File: Tutorial8-1DaggerHilt-Basics/src/main/java/com/smarttoolfactory/tutorial8_1daggerhilt_basics/model/ToastMaker.kt
package com.smarttoolfactory.tutorial8_1daggerhilt_basics.model

import android.content.Context
import android.widget.Toast

class ToastMaker(private val context: Context) {

    fun showToast(message: String) {
        println("🔥 ToastMaker showToast() context: $context")
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

}